libbpfgo/libbpfgo.go

```diff
@@ -302,6 +302,39 @@ func (m *Module) GetMap(mapName string) (*BPFMap, error) {
 	}, nil
 }
 
+func (b *BPFMap) Pin(pinning string) error {
+	cs := C.CString(b.name)
+	path := C.CString(pinning)
+	bpfMap := C.bpf_object__find_map_by_name(b.module.obj, cs)
+	errC := C.bpf_map__pin(bpfMap, path)
+	if errC != 0 {
+		return fmt.Errorf("Failed to pin map %s to path %s", b.name, pinning)
+	}
+	return nil
+}
+
+func (b *BPFMap) Unpin(pinning string) error {
+	cs := C.CString(b.name)
+	path := C.CString(pinning)
+	bpfMap := C.bpf_object__find_map_by_name(b.module.obj, cs)
+	errC := C.bpf_map__unpin(bpfMap, path)
+	if errC != 0 {
+		return fmt.Errorf("Failed unpin path for map %s", b.name)
+	}
+	return nil
+}
+
+func (b *BPFMap) SetPinPath(pinning string) error {
```

Comment:
looks like this does the same as unpin? what did I miss here?

Reply:
You did not miss anything, it was my fault, i did not push the change after testing

Reply:
Would "pinPath" be a suitable name?

Reply:
I have freed pointer and changed the "pinning" to "pinPath"